From 93724aa28fc20c8b7f0167201d1759b7118ba890 Mon Sep 17 00:00:00 2001
From: Alan Pevec <apevec@redhat.com>
Date: Thu, 7 Apr 2011 01:53:51 +0200
Subject: [PATCH] dmsquash-live: hide plymouth while checking ISO

Fixes long-standing FIXME
Latest isomd5sum added an option to abort media check with ESC key,
but that key is taken by plymouth for switching to the detailed log
messages, making it impossible to abort checkisomd5.
Tested in text mode.
---
 modules.d/90dmsquash-live/dmsquash-live-root |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

--- modules.d/90dmsquash-live/dmsquash-live-root.orig	2011-07-22 02:02:58.000000000 +0200
+++ modules.d/90dmsquash-live/dmsquash-live-root	2011-07-22 02:07:20.873266232 +0200
@@ -23,18 +23,19 @@
 getarg readonly_overlay && readonly_overlay="--readonly" || readonly_overlay=""
 overlay=$(getarg overlay)
 
-# FIXME: we need to be able to hide the plymouth splash for the check really
 [ -e $livedev ] && fs=$(blkid -s TYPE -o value $livedev)
 if [ "$fs" = "iso9660" -o "$fs" = "udf" ]; then
     check="yes"
 fi
 getarg check || check=""
 if [ -n "$check" ]; then
+    [ -x /bin/plymouth ] && /bin/plymouth --hide-splash
     checkisomd5 --verbose $livedev || :
     if [ $? -ne 0 ]; then
 	die "CD check failed!"
 	exit 1
     fi
+    [ -x /bin/plymouth ] && /bin/plymouth --show-splash
 fi
 
 getarg ro && liverw=ro
